src/ui/QuerySuggestPreview/QuerySuggestPreview.ts

```diff
@@ -1,59 +1,272 @@
 
-  constructor(public element: HTMLElement, public options?: IQuerySuggestPreview, public bindings?: IComponentBindings) {
-    super(element, options, bindings, QuerySuggestPreview.ID);
+  private previousSuggestionHovered: string;
+  private renderer: ResultListRenderer;
+  private timer;
+  public currentlyDisplayedResults: IQueryResult[] = [];
 
-    //Have to do this for now or or break the search interface
-    //Will fix in JSUI-2508 where i will use my template
-    this.element.style.display = 'none';
+  /**
+   * Creates a new QuerySuggestPreview component.
+   * @param element The HTMLElement on which to instantiate the component.
+   * @param options The options for the QuerySuggestPreview component.
+   * @param bindings The bindings that the component requires to function normally. If not set, these will be
+   * automatically resolved (with a slower execution time).
+   */
+  constructor(public element: HTMLElement, public options?: IQuerySuggestPreview, public bindings?: IComponentBindings) {
+    super(element, QuerySuggestPreview.ID, bindings);
 
     this.options = ComponentOptions.initComponentOptions(element, QuerySuggestPreview, options);
 
+    if (!this.options.resultTemplate) {
+      this.logger.warn(
+        `Option "resultTemplate" is *REQUIRED* on the component "QuerySuggestPreview". The component or the search page might *NOT WORK PROPERLY*`,
+        `Check the following URL to create your result template`,
+        `https://developers.coveo.com/x/aIGfAQ`
+      );
+    }
+
     this.bind.onRootElement(OmniboxEvents.querySuggestGetFocus, (args: IQuerySuggestSelection) => this.querySuggestGetFocus(args));
+    this.bind.onRootElement(OmniboxEvents.querySuggestRendered, () => {
+      this.handleAfterComponentInit();
+    });
+    this.bind.onRootElement(OmniboxEvents.querySuggestLooseFocus, () => {
+      this.handleQuerySuggestLooseFocus();
+    });
+  }
+
+  /**
+   * Gets the list of currently displayed result.
+   * @returns {IQueryResult[]}
+   */
+  public get displayedResults(): IQueryResult[] {
+    return this.currentlyDisplayedResults;
+  }
+
+  private get templateToHtml() {
+    const templateToHtmlArgs: ITemplateToHtml = {
+      searchInterface: this.searchInterface,
+      queryStateModel: this.queryStateModel,
+      resultTemplate: this.options.resultTemplate
+    };
+    return new TemplateToHtml(templateToHtmlArgs);
+  }
+
+  private handleQuerySuggestLooseFocus() {
+    clearTimeout(this.timer);
+    this.timer = null;
+    this.previousSuggestionHovered = null;
+    this.currentlyDisplayedResults = [];
+  }
+
+  private updatePreviewContainer(container: HTMLElement) {
+    if (!this.options.previewWidth) {
+      return;
+    }
+    container.style.width = `${this.options.previewWidth}px`;
+  }
+
+  private updateWidthOfSuggestionContainer(container: HTMLElement) {
+    if (!this.options.suggestionWidth) {
+      return;
+    }
+    container.style.minWidth = this.options.suggestionWidth;
```

Comment:
With not simply set `width` (instead of min and max to the same value ?)

Reply:
Ok so while testing the option, I found out that if the value was a `%`, my container which have the QuerySuggest and the preview always resized for some reason 

I think this is because when we have a long querySuggest, we cut the word somewhere and add `...`. This seem so add a bit to the size and since the size of the container is set in `%`, it will then add to excess width to the preview container resizing it too

For now, I will change the suggestion width to only be in `px` since that way, we don't have the problem.
If we ever receive a request to support `%` we can spend more time resolving the issue